app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java

```diff
@@ -1,76 +1,41 @@
 package fr.free.nrw.commons;
 
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.view.PagerAdapter;
 import android.support.v4.view.ViewPager;
+import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 
 import com.viewpagerindicator.CirclePageIndicator;
 
+import butterknife.BindView;
+import butterknife.ButterKnife;
 import fr.free.nrw.commons.theme.BaseActivity;
 
 public class WelcomeActivity extends BaseActivity {
-    static final int PAGE_WIKIPEDIA = 0,
-            PAGE_DO_UPLOAD = 1,
-            PAGE_DONT_UPLOAD = 2,
-            PAGE_IMAGE_DETAILS = 3,
-            PAGE_FINAL = 4;
-    static final int[] pageLayouts = new int[] {
-            R.layout.welcome_wikipedia,
-            R.layout.welcome_do_upload,
-            R.layout.welcome_dont_upload,
-            R.layout.welcome_image_details,
-            R.layout.welcome_final
-    };
+    private WelcomePagerAdapter adapter;
 
-    private ViewPager pager;
-    private Button yesButton;
+    @BindView(R.id.welcomePager) ViewPager pager;
+    @BindView(R.id.welcomePagerIndicator) CirclePageIndicator indicator;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_welcome);
+        View view = LayoutInflater.from(this).inflate(R.layout.activity_welcome, null);
+        setContentView(view);
 
         getSupportActionBar().hide();
+        ButterKnife.bind(this, view);
```

Comment:
Why not just pass in a reference to the current Activity instead of the view i.e. `ButterKnife.bind(this)`? By doing this you can remove the [instantiation of the `view`](https://github.com/commons-app/apps-android-commons/pull/481/commits/9649bfc78503e1dfafc1730035ce31c850581bb0#diff-44ba9a8fa5ca2b5fd0c82101150c1f04R270) and pass in `R.layout.activity_welcome` to `setContentView`.